pilot/cmd/pilot-agent/config.go

```diff
@@ -85,13 +101,55 @@ func constructProxyConfig() (meshconfig.ProxyConfig, error) {
 	if err := validation.ValidateProxyConfig(&proxyConfig); err != nil {
 		return meshconfig.ProxyConfig{}, err
 	}
-	annotations, err := readPodAnnotations()
-	if err != nil {
-		log.Warnf("failed to read pod annotations: %v", err)
-	}
 	return applyAnnotations(proxyConfig, annotations), nil
 }
 
+// getMeshConfig gets the mesh config to use for proxy configuration
+// 1. First we take the default config
+// 2. Then we apply any settings from file (this comes from gateway mounting configmap)
+// 3. Then we apply settings from environment variable (this comes from sidecar injection sticking meshconfig here)
+// 4. Then we apply overrides from annotation (this comes from annotation on gateway, passed through downward API)
+func getMeshConfig(fileOverride, annotationOverride string) (meshconfig.MeshConfig, error) {
+	mc := mesh.DefaultMeshConfig()
+
+	if fileOverride != "" {
```

Comment:
You inverted the order from the old code; I don't think it matters as we shouldn't have file & env both set in any one workload but why did you change it?

Reply:
The previous logic was not merging all settings, it was env OR file. Arguable we could do the same here but it seemed simpler to have everything just applying on top of the previous. As you mentioned the order doesn't matter care too much which one goes first - I think file < env < annotation just made more sense to me, I can change it if preferred

Reply:
Well, this is not a user facing API but implementation so we should be free to change it. Keep it the way you have, it's cleaner.